solana/rpc/websocket.py

```diff
@@ -0,0 +1,162 @@
+"""API WebSocket Client to interact with the Solana JSON RPC Endpoint."""
+from __future__ import annotations
+
+from typing import Union
+
+from solana.publickey import PublicKey
+
+from .providers.ws import Provider
+from .types import RPCMethod, RPCResponse
+
+
+class WebSocketClient:
+    """Websocket client class."""
+
+    def __init__(self, endpoint: str):
+        """Init API client for websocket."""
+        self._provider = Provider(endpoint)
+
```

Comment:
I think we also need to expose the WebSocket connection here b/c right now this client only allows you to subscribe/unsubscribe. You'll need to double-check this but I think we need something like:

```python
    def __init__(self, endpoint: str):
        """Init API client for websocket."""
        self._provider = Provider(endpoint)
        self._conn = _provider.conn

   def connection(self) -> PersistentWebSocket:
        return self._conn
```

Reply:
https://websockets.readthedocs.io/en/stable/api.html#websockets.client.connect

Reply:
I believe this particular implementation enters and exits the connection whenever a request is made so this method will always return None.